File: GuardianMonsters/battle/src/de/limbusdev/guardianmonsters/battle/model/ObservableList.java
package de.limbusdev.guardianmonsters.battle.model;

/**
 * Created by georg on 04.12.16.
 */

public interface ObservableList<T> {
    void addObserver(ListObserver o);
    void clearObservers();
    void notifyObserversRemove(T item);
    void notifyObserversAdd(T item);
    void notifyObserversSort();
}

File: GuardianMonsters/battle/src/de/limbusdev/guardianmonsters/battle/model/ObservableList.kt
package de.limbusdev.guardianmonsters.battle.model

/**
 * Created by georg on 04.12.16.
 */

interface ObservableList<T>
{
    fun addObserver(o: ListObserver<T>)
    fun clearObservers()
    fun notifyObserversRemove(item: T)
    fun notifyObserversAdd(item: T)
    fun notifyObserversSort()
}
